R/03-miRNAPathwayEnrichment.R

```diff
@@ -1,101 +1,105 @@
-    out.dir <- paste0(out.dir, '/')
-  if (!dir.exists(out.dir))
-    stop('Output directory does not exist.')
-  
-  # select pathways with minimum set size
-  paths.sel  <- sapply(pathways.sets,length)
-  pathways.sets   <- pathways.sets[paths.sel > min.path.size]
-  paths.ref <- Reduce(union,pathways.sets)
-
-  # select miRNAs with targets in pathways
-  mir.sets <- lapply(mir.sets, function(X){X[X %in% paths.ref]})
-  
-  # select pathways with selected genes of interest
-  if (!is.null(genes.selection)){
-    gene.df     <- bitr(genes.selection, fromType = from.id,
-                        toType = to.id,
-                        OrgDb = org.Hs.eg.db)
-    pathways.sets <- lapply(pathways.sets, function(X){
-      X[X %in% gene.df[,c(to.id)]]})
-    
-    mir.sets <- lapply(mir.sets, function(X){
-      X[X %in% gene.df[,c(to.id)]]})
-    paths.ref <- Reduce(union,pathways.sets)
-  }
-  
-  # select miRNAs of interest
-  if (!is.null(mir.selection)){
-    
-    mir.sets <- mir.sets[names(mir.sets) %in% mir.selection]
-    
-  }
-  
-  sel.vec <- sapply(mir.sets,length)
-  mir.sets <- mir.sets[sel.vec > min.path.size]
-  iterator <- (merge(names(mir.sets),names(pathways.sets)))
-  iterator <- iterator %>% mutate_all(.,as.character)
-  all <- length(paths.ref)
-  
-  # find enrichment p-value of each miRNA target set and each pathway set
-  enrichs   <- mclapply(1:nrow(iterator), 
-                        function(Y){
-                          X <- iterator[Y,]
-                          q <- length(intersect(unlist(pathways.sets[X[[2]]]), 
-                                                unlist(mir.sets[X[[1]]])))  
-                          m <- length(unlist(mir.sets[X[[1]]]))
-                          n <- all - m
-                          k <- length(unlist(pathways.sets[X[[2]]]))
-                          pval <- phyper(q-1,m,n,k,lower.tail = F,log.p = F)
-                          return(c("pval" = pval,
-                                   "Intersect"    = q, 
-                                   "mirset_Size"  = m,
-                                   "not_mirset"   = n, 
-                                   "pathway_Size" = k,
-                                   "ratio_in"     = q/(k-q),
-                                   "ratio_out"    = (m-q)/n,
-                                   "ratio_ratios" = (q/(k-q))/((m-q)/n) ))
-                        }, mc.cores = num.cores)
-  
-  temp      <- do.call(rbind,enrichs)
-  temp      <- as.data.frame(temp)
-  iterator  <- cbind(iterator,temp)
-  
-  if (!is.null(save.RDS.name)){
-    saveRDS(iterator, paste0(out.dir, save.RDS.name))
-  }
-  return (iterator)
-}
-
+#' @description Outputs enrichment probability of miRNAs based on 
+#' pathway clusters
+#' @param mir.sets a table of miRNAs and a list of their interactions with
+#'  genes in ENTREZ ID 
+#' @param pathways.sets a table of pathways and a list of their interactions
+#'       with genes in ENTREZ ID
```

Comment:
"If any description corresponding to a roxygen tag spans over multiple lines, add another two spaces of extra indention."
https://style.tidyverse.org/documentation.html#indents-and-line-breaks